galgebra/lt.py

```diff
@@ -754,14 +754,14 @@ def __init__(self, f, Ga, nargs=None, fct=False):
 
         else:
             if isinstance(f, types.FunctionType):  # Tensor defined by general multi-linear function
-                args, _varargs, _kwargs, _defaults = inspect.getargspec(f)
+                args = inspect.getfullargspec(f)[0]
                 self.nargs = len(args)
                 self.f = f
                 Mlt.increment_slots(self.nargs, Ga)
                 self.fvalue = f(*tuple(Ga._mlt_a[0:self.nargs]))
             else:  # Tensor defined by component expression
                 self.f = None
-                self.nargs = len(args)
+                self.nargs = 0
                 Mlt.increment_slots(self.nargs, Ga)
                 self.fvalue = f
```

Comment:
I'd be tempted to replace this with `raise NotImplementedError` (or just leave it broken), as otherwise I have no idea what this code is supposed to do. It clearly has never been tested.

Reply:
Agreed